Remove modAction and legacy controller support

Mark-H opened the pull request:
### What does it do?

Removing modAction and relevant legacy, which was deprecated in v2.2. Modernise the modManagerResponse and controller interaction. Improve stuff.

### Why is it needed?

Code cleanup. 

### Related issue(s)/PR(s)

Documentation update: https://github.com/modxorg/Docs/pull/160

### To do

- [x] Fix login; currently the modManagerResponse handling is _too_ strict and it wont show the login page
- [x] Figure out what to do with setup/includes/upgrade.install.php
- [x] Add tests to prove the class name handling is correct and can find classes 
- [x] Add removed files to the cleanup migration script 
- [x] Improve error pages
- [x] Check if actions lexicon can be removed, and if so, remove it 
- [x] Remove relevant settings - cache_action_map, others?
- [ ] (optional) Squash all commits either in PR or as part of merge
- [x] Remove modManagerControllerDeprecated

Mark-H committed:
Start removing modAction and relevant legacy code

Mark-H committed:
Fix ConfigJs processor using invalid $scriptProperties, always set namespace

Ruslan-Aleev commented:
Perhaps it’s worth deleting lexicons as well - https://github.com/modxcms/revolution/blob/3.x/core/lexicon/en/action.inc.php

Mark-H committed:
Remove more references to actions

Mark-H committed:
Prettify errors with a special Error controller, to render inside the standard manager panel

Mark-H committed:
Fix inability to login, pass error information via constructor, remove validateAuthentication

Mark-H committed:
More simplification of modManagerResponse and permission checking

Mark-H committed:
Catch any \Exception and \Error triggered in a controller

JoshuaLuckers reviewed (commented):
I see `file_exists` and `is_readable` in the same if statements. `is_readable` checks if the file exists and is readable so there is no need to check both.

JoshuaLuckers commented on core/src/Revolution/modManagerResponse.php:
```diff
@@ -247,30 +213,74 @@ public function checkForMenuPermissions($action)
+        }
+
+        // If the file exists, require it, while discarding any content from it
+        if (file_exists($fullPath) && is_readable($fullPath)) {
```
```suggestion
        if (is_readable($fullPath)) {
```
> is_readable — Tells whether a file exists and is readable

JoshuaLuckers commented:
Is `modManagerControllerDeprecated` being removed by this PR as well?

Mark-H committed:
Pass the namespace_path along to the controller

Mark-H committed:
Change signature for modManagerController::_construct to define visibility and no pass-by-ref

Mark-H committed:
Use strict typing for modManagerResponse::checkForMenuPermissions

Mark-H committed:
Fix unit tests failing locally due to $this->modx->controller not being set by test harnass

Mark-H committed:
Add some tests for loading controllers from action names, fix __construct signature on modResponse

Mark-H committed:
Cleanup files on upgrade

Mark-H committed:
Remove modManagerControllerDeprecated

Mark-H committed:
Remove cache_action_map setting and references to action_map cache

Mark-H committed:
Remove unused action lexicons

Most of these had not been in use since 2.3 or so with the introduction of proper controllers. The remaining are still used in menu management.

Mark-H committed:
Remove legacy upgrade code from 2.0.5, no longer relevant

Mark-H commented:
I think this should now be ready for review and testing. Documentation updates are pending at https://github.com/modxorg/Docs/pull/160, though there are probably more references to modAction in the docs that should be updated.

Mark-H commented:
@cla-bot check

JoshuaLuckers commented:
I'm not too familiar with `modAction*` classes and I was wondering how the classes below are related to `modAction` and if they should stay or not:
- modAccessActionDom 
- modActionDom
- modActionField

Mark-H commented:
Iirc, modActionDom is the "regions" tab in form customization. modActionField the individual fields you can enable/move. Don't think modAccessActionDom is implemented, but supposed to be related to the regions.

opengeek commented:
Any chance you can squash this PR and force push @Mark-H ?
